### PR TITLE
MINOR: Include additional detail in fetch error message

### PR DESCRIPTION
**Pull request by hachikuji**
Issues such as KAFKA-7656 would be easier to debug if the error message included additional detail about the fetch request.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by huangyiminghappy** [comment]
https://github.com/apache/kafka/pull/6002 @hachikuji  could you help me review the pr? thank you

File: core/src/main/scala/kafka/api/Request.scala [hunk 1]
```diff
@@ -24,4 +24,14 @@ object Request {
 
   // Broker ids are non-negative int.
   def isValidBrokerId(brokerId: Int): Boolean = brokerId >= 0
+
+  def describeReplicaId(replicaId: Int): String = {
+    replicaId match {
+      case OrdinaryConsumerId => "consumer"
+      case DebuggingConsumerId => "debug consumer"
+      case FutureLocalReplicaId => "future local replica"
+      case id if isValidBrokerId(id) => s"replica [$replicaId]"
```
Comment (mumrah) [inline core/src/main/scala/kafka/api/Request.scala]: Doesn't really matter, but you have `$replicaId` here and `$id` below

File: core/src/main/scala/kafka/server/ReplicaManager.scala [hunk 2]
```diff
@@ -946,7 +946,11 @@ class ReplicaManager(val config: KafkaConfig,
         case e: Throwable =>
           brokerTopicStats.topicStats(tp.topic).failedFetchRequestRate.mark()
           brokerTopicStats.allTopicsStats.failedFetchRequestRate.mark()
-          error(s"Error processing fetch operation on partition $tp, offset $offset", e)
+
+          val fetchSource = Request.describeReplicaId(replicaId)
+          error(s"Error processing fetch with max size $adjustedMaxBytes from $fetchSource " +
+            s"on partition $tp: $fetchInfo", e)
```
Comment (mumrah) [inline core/src/main/scala/kafka/server/ReplicaManager.scala:952]: Offset is not longer included in this error message. Is that intentional?

Comment (hachikuji) [reply core/src/main/scala/kafka/server/ReplicaManager.scala:952]: It's part of `PartitionData`.

**Review by mumrah — APPROVED** [review]
LGTM :shipit:

**Pull request merged**